### PR TITLE
Changes to accommodate configurations updates

### PR DESCRIPTION
**Pull request by ashwin1111**

### CODE REVIEW

File: src/app/core/models/updated-configuration.ts [hunk 1]
```diff
@@ -12,5 +12,6 @@ export type UpdatedConfiguration = {
     oldValue: string,
     newValue: string
   },
-  autoCreateDestinationEntity: boolean
+  autoCreateDestinationEntity: boolean,
+  acceptedChanges?: boolean
```
Comment (labhvam5) [inline src/app/core/models/updated-configuration.ts]: why is there a '?'

File: src/app/netsuite/settings/netsuite-configurations/configuration/configuration.component.ts [hunk 2]
```diff
@@ -256,11 +257,15 @@ export class ConfigurationComponent implements OnInit {
...
     });
+    const trackingProperties = updatedConfigurations;
+    trackingProperties.acceptedChanges = false;
 
     dialogRef.afterClosed().subscribe(data => {
       if (data.accpetedChanges) {
         that.postConfigurationsAndMappingSettings(generalSettingsPayload, mappingSettingsPayload, true, data.redirectToEmployeeMappings);
+        trackingProperties.acceptedChanges = true;
       }
+      that.trackingService.onUpdateConfiguration(trackingProperties);
```
Comment (labhvam5) [inline src/app/netsuite/settings/netsuite-configurations/configuration/configuration.component.ts:271]: should it be in the if block ?

File: src/app/core/models/updated-configuration.ts (hunk 1 above)
Comment (ashwin1111) [reply src/app/core/models/updated-configuration.ts]: Have removed this in the next commit.
To answer your question, when this is an optional key we add ? to it.

File: src/app/netsuite/settings/netsuite-configurations/configuration/configuration.component.ts (hunk 2 above)
Comment (ashwin1111) [reply src/app/netsuite/settings/netsuite-configurations/configuration/configuration.component.ts:271]: This is fine. We are tracking both update as well as closing the popup without hitting save. Would help us understand better about the behaviour. `acceptedChanges` would be true/false.

**Pull request merged**